kernel/api/file.go

```diff
@@ -172,9 +172,18 @@ func readDir(c *gin.Context) {
 
 	files := []map[string]interface{}{}
 	for _, entry := range entries {
+		path := filepath.Join(dirPath, entry.Name())
+		info, err = os.Stat(path)
+		if nil != err {
+			logging.LogErrorf("stat [%s] failed: %s", path, err)
+			ret.Code = 500
+			ret.Msg = err.Error()
+			return
+		}
 		files = append(files, map[string]interface{}{
-			"name":  entry.Name(),
-			"isDir": entry.IsDir(),
+			"name":      entry.Name(),
+			"isDir":     info.IsDir(),
```

Comment:
```suggestion
			"isDir":     entry.IsDir(),
```
Why change from `entry.IsDir()` to `info.IsDir()` ?
`entry.IsDir()` / `dir.IsDir()` (where `entry` and `dir` are of type `fs.DirEntry`) is being used in `kernel/util/file.go > IsDirRegularOrSymlink` too.

Reply:
```go
// entry is a symlink to a dir
path := filepath.Join(dirPath, entry.Name())
info, err = os.Stat(path)

entry.IsDir() // return false
info.IsDir() // return true
```

Reply:
Guys, let's make sure the code here is ok:

![image](https://github.com/siyuan-note/siyuan/assets/873584/2f6a8187-1b73-4c5f-9c0c-037ca80be72d)

Reply:
Oh, sorry, I missed the changes in file.go later, I have extracted a function `IsSymlink()`, it seems to be no problem.

Reply:
> Guys, let's make sure the code here is ok:
> 
> ![image](https://user-images.githubusercontent.com/873584/255333496-2f6a8187-1b73-4c5f-9c0c-037ca80be72d.png)

If this function's name is `IsDirRegularOrSymlink`, it's Ok.
If this function's name is `IsDirRegularOrDirSymlink`, it's not Ok.👀

Reply:
@Zuoqiu-Yingyi Thanks for the explanation, I get it, merge first.

@fenilgmehta If there is something we missed, please continue to add comments, thank you.